syntax = "proto3";

package main;


message putRequest {
  string key = 1;
  string value = 2;
  int32 sequenceNumber = 3;
} 

message getRequest {
  string key = 1;
} 


message replyPutRequest {
  string response  = 1;
} 

message replyGetRequest {
  string response  = 1;
} 


service PrimaryBackupService {
  rpc putRequestOperation(putRequest) returns (replyPutRequest);
  rpc getRequestOperation(getRequest) returns (replyGetRequest);
} 
